apps/api/src/campaign-file/campaign-file.controller.ts

```diff
@@ -13,36 +13,43 @@ import {
 } from '@nestjs/common'
 import { FilesInterceptor } from '@nestjs/platform-express'
 import { UseInterceptors, UploadedFiles } from '@nestjs/common'
-import { Public, AuthenticatedUser, RoleMatchingMode, Roles } from 'nest-keycloak-connect'
-import { RealmViewSupporters, ViewSupporters } from '@podkrepi-bg/podkrepi-types'
+import { Public, AuthenticatedUser } from 'nest-keycloak-connect'
 import { PersonService } from '../person/person.service'
 import { FilesRoleDto } from './dto/files-role.dto'
 import { CampaignFileService } from './campaign-file.service'
+import { CampaignService } from '../campaign/campaign.service'
+import { KeycloakTokenParsed, isAdmin } from '../auth/keycloak'
 
 @Controller('campaign-file')
 export class CampaignFileController {
   constructor(
     private readonly campaignFileService: CampaignFileService,
     private readonly personService: PersonService,
+    private readonly campaignService: CampaignService,
   ) {}
 
   @Post(':campaign_id')
   @UseInterceptors(FilesInterceptor('file'))
-  @Roles({
-    roles: [RealmViewSupporters.role, ViewSupporters.role],
-    mode: RoleMatchingMode.ANY,
-  })
   async create(
     @Param('campaign_id') campaignId: string,
     @Body() body: FilesRoleDto,
     @UploadedFiles() files: Express.Multer.File[],
-    @AuthenticatedUser() user,
+    @AuthenticatedUser() user: KeycloakTokenParsed,
   ) {
-    const person = await this.personService.findOneByKeycloakId(user.sub)
+    const id = user.sub as string
+    const person = await this.personService.findOneByKeycloakId(id)
     if (!person) {
-      Logger.warn('No person record with keycloak ID: ' + user.sub)
-      throw new NotFoundException('No person record with keycloak ID: ' + user.sub)
+      Logger.warn('No person record with keycloak ID: ' + id)
+      throw new NotFoundException('No person record with keycloak ID: ' + id)
     }
+
+    if (!isAdmin(user)) {
+      const campaign = await this.campaignService.getCampaignByIdAndPersonId(campaignId, id)
```

Comment:
Adding a small note: Here we pass `id` which is `keycloakId` but the function accepts `personId` which is different ID